arrow/memory/allocator.go

```diff
@@ -25,3 +25,58 @@ type Allocator interface {
 	Reallocate(size int, b []byte) []byte
 	Free(b []byte)
 }
+
+// AlignedAllocator is an Allocator that returns buffers aligned to 64-byte boundaries.
```

Comment:
any particular reason to pick 64-byte boundary as opposed to letting it be configurable?